alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/main/java/org/alfresco/transformer/AIOController.java

```diff
@@ -0,0 +1,221 @@
+    @Override
+    public ProbeTestTransform getProbeTestTransform() 
+    {
+        // HtmlParserContentTransformer html -> text
+        // See the Javadoc on this method and Probes.md for the choice of these values.
+        return new ProbeTestTransform(this, "quick.html", "quick.txt",
+            119, 30, 150, 1024,
+            60 * 2 + 1, 60 * 2)
+        {
+            @Override
+            protected void executeTransformCommand(File sourceFile, File targetFile)
+            {
+                Map<String, String> parameters = new HashMap<>();
+                parameters.put(AllInOneTransformer.TRANSFORM_NAME_PARAMETER, "misc");
+                parameters.put(SOURCE_ENCODING, "UTF-8");
+                try
+                {
+                    transformer.transform(sourceFile, targetFile, MIMETYPE_HTML,
+                    MIMETYPE_TEXT_PLAIN, parameters);
+                }
+                catch(Exception e)
+                {
+                    logger.error(e.getMessage(), e);
+                }
+                
+            }
+        };
+    }
+    
+    @PostMapping(value = "/transform", consumes = MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<Resource> transform(HttpServletRequest request,
+        @RequestParam("file") MultipartFile sourceMultipartFile,
+        @RequestParam(TARGET_EXTENSION) String targetExtension,
+        @RequestParam(TARGET_MIMETYPE) String targetMimetype,
+        @RequestParam(SOURCE_MIMETYPE) String sourceMimetype,
+        @RequestParam Map<String, String> transformOptions,
+        @RequestParam (value = TEST_DELAY, required = false) Long testDelay)
+    {   
+        //Using @RequestParam Map<String, String> will gather all text params, including those specified seperately above.
+        removeUnwantedOptions(transformOptions, UNWANTED_OPTIONS, true);
+
+        final String targetFilename = createTargetFileName(
+            sourceMultipartFile.getOriginalFilename(), targetExtension);
+        getProbeTestTransform().incrementTransformerCount();
+        final File sourceFile = createSourceFile(request, sourceMultipartFile);
+        final File targetFile = createTargetFile(request, targetFilename);
+
+        // TODO Currently sourceMimetype and targetMimetype could be an empty string how does this affect getting the name?
+        // not all controllers take these from the request? Do requests intended for these transforms provide these?
+        final String transform = getTransformerName(sourceFile, sourceMimetype, targetMimetype, transformOptions);
+
+        transformOptions.put(AllInOneTransformer.TRANSFORM_NAME_PARAMETER, transform);
+        debugLogTransform(sourceMimetype, transformOptions);
+
+        try 
+        {
+            transformer.transform(sourceFile, targetFile, sourceMimetype, targetMimetype, transformOptions);
+        } 
+        catch (Exception e) 
+        {
+            logger.error(e.getMessage(), e);

```

Comment:
Shouldn't a transform exception return an http 4xx exception back to the requestor - not success ?

Reply:
Note: I think @PunchedChimera made some specific changes in https://github.com/Alfresco/alfresco-transform-core/pull/208 that could ben cherry-picked here ?

Reply:
Has been fixed in https://github.com/Alfresco/alfresco-transform-core/pull/200/commits/223212461733bad6929bdaba500e2fb19b3e8ba2